lib/libc/Kconfig

```diff
@@ -43,6 +44,7 @@ config PICOLIBC
 	select THREAD_LOCAL_STORAGE if ARCH_HAS_THREAD_LOCAL_STORAGE && TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE
 	select LIBC_ERRNO if THREAD_LOCAL_STORAGE
 	depends on !NATIVE_APPLICATION
+	depends on PICOLIBC_SUPPORTED
```

Comment:
I'd left this out as external toolchains may not indicate picolibc support? How do we detect newlib support for those?

Reply:
In Kconfig ? we don't.
But we should definately improve on this, and if we have a picolibc helper symbol then I believe we should use it.
Alternatively, if it is not actively used, then remove it and place full responsibility at the end-user (like it's done for newlib).
My preference is to improve things, and when changing picolibc handling, then that's a good place to start.

We do have the `TOOLCHAIN_HAS_NEWLIB`, but that is only used for twister test filtering.
It's set here:
https://github.com/zephyrproject-rtos/zephyr/blob/ef203fd8aeb71758a106c4416aff5effba42c8e2/cmake/toolchain/xtools/generic.cmake#L36
https://github.com/zephyrproject-rtos/zephyr/blob/ef203fd8aeb71758a106c4416aff5effba42c8e2/cmake/toolchain/gnuarmemb/generic.cmake#L21

but was intended for twister filtering, as seen in #12393
would probably be good to cleanup this part and make it a bit more consistent.

Reply:
created: #53856